libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/market-roles/market-role-group.service.ts

```diff
@@ -22,24 +39,34 @@ import { EditableMarketRoleRow } from './dh-market-participant-actor-market-role
 @Injectable()
 export class MarketRoleGroupService {
   readonly groupRows = (rows: EditableMarketRoleRow[]) => {
-    const marketRolesMap = rows.reduce((map, row) => {
-      map.set(row.marketRole, [...(map.get(row.marketRole) || [])]);
-      if (row.gridArea)
-        map.set(row.marketRole, [
-          ...map.get(row.marketRole),
-          {
-            id: row.gridArea,
-            meteringPointTypes: row.meteringPointTypes,
-          },
-        ]);
-      return map;
-    }, new Map());
-
     const result: MarketRole[] = [];
 
-    marketRolesMap.forEach((v, k) =>
-      result.push({ marketRole: k, gridAreas: v })
-    );
+    for (let index = 0; index < rows.length; index++) {
+      const row = rows[index];
+      const gridAreaId = row.gridArea;
+
+      const gridArea = {
+        id: gridAreaId!,
```

Comment:
issue: the non-null assertion here is incorrect because the `gridArea` is `null` when a new row is added. So typing the property as `string` is partly true.

![image](https://user-images.githubusercontent.com/1096332/200563677-d78777c1-bd7c-49a6-b8d6-51f538bf0393.png)

suggestion: revisit the non-null assertion here and in other places in this file to make sure the proper types are used.

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1114/commits/4069d56e668a1cfeac17e111e7f1ab4d9d487175